spec/runner_spec.rb

```diff
@@ -100,39 +95,83 @@
 
   describe '#run!' do
 
-      context 'with malformed queries' do
-        Fixture.get(:syntax_error).each do | fixture |
-          context "'#{fixture['content']}'" do
+    shared_examples_for 'a correct query' do |exercise, query|
+      let(:result) { run exercise, query }
+
+      it { expect(result[0][0][1]).to eq :passed }
```

Comment:
Perhaps you could try using the `json_eq` and `json_like` matchers - https://github.com/mumuki/mumukit-core/blob/e51bf5e52120ece9ceb8a89fa581922f22783af2/lib/mumukit/core/rspec.rb -, so that you don't to do thinks like `result[0][0][1]` but instead testing the whole output. 

We use them in many runners in order to check json outputs with more precision but less boilerplate.